### ISSUE TITLE
Fix most instances of extension scanner warnings for TYPO3 7, 8, 9

### ISSUE THREAD

**Issue by oliverklee** [oliverklee/ext-oelib#416]
Fix the issues in code that will not be removed in oelib 4.0 anyway. We might need version-dependent code here (with version switches). For things that require a bigger refactoring, we need separate tickets, and we might need to move them to the 4.0 milestone.

We might split this up into multiple PRs.

Blocked by #387.
